pinot-core/src/test/java/com/linkedin/pinot/core/segment/index/creator/SegmentPartitionTest.java

```diff
@@ -178,6 +178,62 @@ public void testPruner() {
   }
 
   /**
+   * Unit test for {@link PartitionSegmentPruner}.
+   * <ul>
+   *   <li> Generates queries with in predicate on partitioned column with random values. </li>
+   *   <li> Ensures that if any column values that are in partition range ([0 5]) do not prune the segment,
+   *        whereas other values do. </li>
+   *   <li> Ensures that predicates on non-partitioned columns do not prune the segment. </li>
+   * </ul>
+   */
+  @Test
+  public void testPrunerIn() {
```

Comment:
They test two different features with different characteristics even though the flow is similar.